Tools/YamlCreate.ps1

```diff
@@ -753,17 +753,18 @@ Function Read-NestedInstaller {
       if ($_EffectiveType -eq 'portable') {
         do {
           Write-Host -ForegroundColor 'Red' $script:_returnValue.ErrorString()
-          Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the portable command alias'
-          if (Test-String -not $_Alias -IsNull) { Write-Host -ForegroundColor 'DarkGray' "Old Variable: $_Alias" }
+          Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the portable command alias'
+          if (Test-String -not "$($_InstallerFile['PortableCommandAlias'])" -IsNull) { Write-Host -ForegroundColor 'DarkGray' "Old Variable: $($_InstallerFile['PortableCommandAlias'])" }
           $_Alias = Read-Host -Prompt 'PortableCommandAlias' | TrimString
           if (Test-String -not $_Alias -IsNull) { $_InstallerFile['PortableCommandAlias'] = $_Alias }
+          elseif (Test-String -not $_InstallerFile['PortableCommandAlias'] -IsNull) { $_InstallerFile['PortableCommandAlias'] = "$($_InstallerFile['PortableCommandAlias'])" }
 
-          if (Test-String $_Alias -MinLength $Patterns.PortableCommandAliasMinLength -MaxLength $Patterns.PortableCommandAliasMaxLength) {
+          if (Test-String $_InstallerFile['PortableCommandAlias'] -MinLength $Patterns.PortableCommandAliasMinLength -MaxLength $Patterns.PortableCommandAliasMaxLength -AllowNull) {
             $script:_returnValue = [ReturnValue]::Success()
           } else {
             $script:_returnValue = [ReturnValue]::LengthError($Patterns.PortableCommandAliasMinLength, $Patterns.PortableCommandAliasMaxLength)
           }
-          if ($_Alias -in @($_NestedInstallerFiles.PortableCommandAlias)) {
+          if ("$($_InstallerFile['PortableCommandAlias'])" -in @($_NestedInstallerFiles.PortableCommandAlias)) {
```

Comment:
This may throw an error if the alias exists and isn't changed, since the `InstallerFile.PortableCommandAlias` could be existing in the portable command aliases already in the `NestedInstallerFiles`.  The validation was on the user input to specifically avoid any possibility of a collision with existing data

Reply:
I'm not sure I understand the concern. Don't we want to stop the user from entering a duplicate in that case?
I'm thinking about the following scenario:
1) User types a duplicate portable alias (one that already is in NestedInstallerFiles from previous portable entry) e.g., `test`. This gets written into InstallerFile as it's not a null input.
2) Alias Collision error thrown. User presses <kbd>Enter</kbd> and *null** value is passed which doesn't get written to InstallerFile. If $_Alias check is used, InstallerFile would hold the value `test`, the script would return success as $_Alias being null is not in NestedInstallerFiles and the value would get written into the manifest. This would result in a manifest that fails `winget validate` with duplicate portable aliases. The check currently used in the PR forces the user to enter a non-duplicate alias.